ci/scripts/utils/rocotostat.py

```diff
@@ -39,60 +41,137 @@ def input_args():
+
+    rocoto_status = {}
+    status_counts = Counter(case for sublist in rocotostat_output for case in sublist)
     for case in status_cases:
-        rocoto_status[case] = sum([sublist.count(case) for sublist in rocotostat_output_all])
+        rocoto_status[case] = status_counts[case]
 
     return rocoto_status
 
 
+def is_done(rocoto_status):
+    """
+    IS_DONE Check if all cycles are done.
+
+    IS_DONE(ROCOTO_STATUS) checks if the total number of cycles equals the number of
+    done cycles in the ROCOTO_STATUS dictionary.
+
+    Input:
+    ROCOTO_STATUS - A dictionary with the count of each status case.
+
+    Output:
+    BOOLEAN - True if all cycles are done, False otherwise.
+    """
+
+    if rocoto_status['CYCLES_TOTAL'] == rocoto_status['CYCLES_DONE']:
+        return True
+    else:
+        return False
+
+
+def is_stalled(rocoto_status):
+    """
+    IS_STALLED Check if all cycles are stalled.
+
+    IS_STALLED(ROCOTO_STATUS) checks if all cycles are stalled by verifying if
+    there are no jobs that are RUNNING, SUBMITTING, or QUEUED.
+
+    Input:
+    ROCOTO_STATUS - A dictionary with the count of each status case.
+
+    Output:
+    BOOLEAN - True if all cycles are stalled, False otherwise.
+    """
+
+    if rocoto_status['RUNNING'] + rocoto_status['SUBMITTING'] + rocoto_status['QUEUED'] == 0:
+        return True
+    else:
+        return False
+
+
 if __name__ == '__main__':
+    """
+    MAIN Execute the script.
+
+    MAIN() parses the input arguments, checks if the rocotostat command is available,
+    adds default arguments to the rocotostat command, and runs it and reports
+    out to stdout spcific information of rocoto workflow.
+    """
 
     args = input_args()
 
+    try:
+        rocotostat = which("rocotostat")
+    except CommandNotFoundError:
+        logger.exception("rocotostat not found in PATH")
+        raise CommandNotFoundError("rocotostat not found in PATH")
+
+    rocotostat.add_default_arg(['-w', os.path.abspath(args.w.name), '-d', os.path.abspath(args.d.name)])
+
     error_return = 0
-    rocoto_status = rocoto_statcount()
+    rocoto_status = rocoto_statcount(rocotostat)
+    rocoto_status.update(rocotostat_summary(rocotostat))
 
-    if rocoto_status['CYCLES_TOTAL'] == rocoto_status['CYCLES_DONE']:
+    if is_done(rocoto_status):
         rocoto_state = 'DONE'
     elif rocoto_status['DEAD'] > 0:
         error_return = rocoto_status['FAIL'] + rocoto_status['DEAD']
         rocoto_state = 'FAIL'
     elif 'UNKNOWN' in rocoto_status:
         error_return = rocoto_status['UNKNOWN']
         rocoto_state = 'UNKNOWN'
-    elif rocoto_status['RUNNING'] + rocoto_status['SUBMITTING'] + rocoto_status['QUEUED'] == 0:
+    elif is_stalled(rocoto_status):
         #
         #  TODO for now a STALLED state will be just a warning as it can
         #  produce a false negative if there is a timestamp on a file dependency.
```

Comment:
Hey look, I remembered why this was giving false negatives 😉. Separate PR, but none of our 'age' requirements are more than 120, so if it is "stalled" for two consecutive checks (assuming they are more than 2 min apart), it should cut out the false positives.

Real solution is to remove all of the file age dependencies, but that will take more development (see also #2349).